src/chat/tools/getConnections.ts

```diff
@@ -0,0 +1,183 @@
+    toolCall: LanguageModelToolCallPart,
+    token: CancellationToken,
+  ): Promise<LanguageModelChatMessage[]> {
+    const parameters = toolCall.input as IGetConnectionsParameters;
+
+    const result: LanguageModelToolResult = await this.invoke(
+      {
+        input: parameters,
+        toolInvocationToken: request.toolInvocationToken,
+      },
+      token,
+    );
+
+    const messages: LanguageModelChatMessage[] = [];
+    if (result.content && Array.isArray(result.content)) {
+      if (this.foundConnectionTypes.length) {
+        let message = new MarkdownString(
+          `Below are the details of available connections for you to reference and summarize to the user:\n\n# Connections`,
+        );
+        for (const part of result.content as LanguageModelTextPart[]) {
+          message = message.appendMarkdown(`\n\n${part.value}`);
+        }
+        messages.push(this.toolMessage(message.value, "result"));
+      }
+
+      if (this.missingConnectionTypes.length) {
+        // summarize missing connection types
+        const missingTypes = this.missingConnectionTypes
+          .map((ctype) => titleCase(ctype))
+          .join(", ");
+        let connectionMessage = new MarkdownString(
+          `The following connection types are not available: ${missingTypes}.`,
+        );
+        messages.push(this.toolMessage(connectionMessage.value, "result"));
+
+        // then add system-message hints about showing buttons for connecting
+        let buttonInstructions = "After explaining the connection status to the user, suggest:";
+        if (this.missingConnectionTypes.includes(ConnectionType.Ccloud)) {
+          buttonInstructions += `\n- They need to sign in to Confluent Cloud using the '${CCLOUD_SIGN_IN_BUTTON_LABEL}' button added above`;
+          const ccloudCommand: Command = {
+            command: "confluent.connections.ccloud.signIn",
+            title: CCLOUD_SIGN_IN_BUTTON_LABEL,
+          };
+          stream.button(ccloudCommand);
+        }
+        if (this.missingConnectionTypes.includes(ConnectionType.Local)) {
+          buttonInstructions +=
+            "\n- They can start local resources using the 'Start Local Resources' button added above";
+          const localCommand: Command = {
+            command: "confluent.docker.startLocalResources",
+            title: "Start Local Resources",
+          };
+          stream.button(localCommand);
+        }
+        if (this.missingConnectionTypes.includes(ConnectionType.Direct)) {
+          buttonInstructions +=
+            "\n- They can connect directly using the 'Add New Connection' button added above";
+          const directCommand: Command = {
+            command: "confluent.connections.direct",
+            title: "Add New Connection",
+          };
+          stream.button(directCommand);
+        }
+        messages.push(systemMessage(buttonInstructions));
+      }
+    }
+
+    // TODO(shoup): remove after debugging
```

Comment:
Just calling out the TODO :)

Reply:
Going to keep these around until the PR stack is done, but will need to do some cleanup once the tools are happy and playing nicely together

Reply:
Moved up to a more generic location in https://github.com/confluentinc/vscode/pull/1717/commits/172da1f5cdbbad06d2c29f2517307a659f519dc8 so we only have one place that needs cleanup later